provider/cmd/leaseLogs.go

```diff
@@ -78,47 +89,65 @@ func doServiceLogs(cmd *cobra.Command) error {
 		return errors.Errorf("tail flag supplied with invalid value. must be >= -1")
 	}
 
-	cert, err := cutils.LoadAndQueryCertificateForAccount(cmd.Context(), cctx, cctx.Keyring)
-	if err != nil {
-		return err
+	type result struct {
+		lid      mtypes.LeaseID
+		provider sdk.Address
+		error    error
+		stream   *gwrest.ServiceLogs
 	}
 
-	gclient, err := gwrest.NewClient(akashclient.NewQueryClientFromCtx(cctx), prov, []tls.Certificate{cert})
-	if err != nil {
-		return err
-	}
+	streams := make([]result, 0, len(leases))
 
-	result, err := gclient.LeaseLogs(cmd.Context(), bid.LeaseID(), svcs, follow, tailLines)
-	if err != nil {
-		return showErrorToUser(err)
+	ctx := cmd.Context()
+
+	for _, lid := range leases {
+		stream := result{lid: lid}
+		prov, _ := sdk.AccAddressFromBech32(lid.Provider)
+		gclient, err := gwrest.NewClient(akashclient.NewQueryClientFromCtx(cctx), prov, []tls.Certificate{cert})
+		if err == nil {
+			stream.stream, stream.error = gclient.LeaseLogs(ctx, lid, svcs, follow, tailLines)
+		} else {
+			stream.error = err
+		}
+
+		streams = append(streams, stream)
 	}
 
-	printFn := func(msg gwrest.ServiceLogMessage) error {
-		fmt.Printf("[%s] %s\n", msg.Name, msg.Message)
-		return nil
+	var wgStreams sync.WaitGroup
+
+	printFn := func(lid mtypes.LeaseID, msg gwrest.ServiceLogMessage) {
+		fmt.Printf("[%s][%s] %s\n", lid, msg.Name, msg.Message)
 	}
 
 	if outputFormat == "json" {
-		printFn = func(msg gwrest.ServiceLogMessage) error {
-			return cmdcommon.PrintJSON(cctx, msg)
+		printFn = func(lid mtypes.LeaseID, msg gwrest.ServiceLogMessage) {
+			type logEntry struct {
+				gwrest.ServiceLogMessage `json:",inline"`
+				Lid                      mtypes.LeaseID `json:"lease_id"`
+			}
+
+			log := &logEntry{
+				ServiceLogMessage: msg,
+				Lid:               lid,
+			}
+			_ = cmdcommon.PrintJSON(cctx, log)
 		}
 	}
 
-	for res := range result.Stream {
-		err = printFn(res)
-		if err != nil {
-			return err
-		}
-	}
+	for _, stream := range streams {
+		if stream.error == nil {
```

Comment:
again please get past the preconditions quickly and without bumping indentation if you can.